### PR TITLE
chore: dependency updates for 0.31

### PR DESCRIPTION
**Pull request by lidel**
- [x]  manual bump of selected ones  for https://github.com/ipfs/kubo/issues/10499
- [x] document new metrics in changelog
- [x] redo this PR to keep go 1.22 and avoid explicit toolkit

### CODE REVIEW

File: test/sharness/t0119-prometheus-data/prometheus_metrics
Comment (lidel) [inline test/sharness/t0119-prometheus-data/prometheus_metrics:1]: :information_source: these are introduced by prometheus update

File: test/sharness/t0119-prometheus-data/prometheus_metrics [hunk 1]
```diff
@@ -94,7 +96,6 @@ go_memstats_heap_objects
 go_memstats_heap_released_bytes
 go_memstats_heap_sys_bytes
 go_memstats_last_gc_time_seconds
-go_memstats_lookups_total
```
Comment (lidel) [inline test/sharness/t0119-prometheus-data/prometheus_metrics:97]: :information_source: Rationale for removal in https://github.com/prometheus/client_golang/releases/tag/v1.20.0 ("was broken anyway")

File: go.mod [hunk 2]
```diff
@@ -251,4 +251,6 @@ require (
 	lukechampine.com/blake3 v1.3.0 // indirect
 )
 
-go 1.22
+go 1.22.0
+
+toolchain go1.23.0
```
Comment (gammazero) [inline go.mod:256]: Do you really want this `toolchain`?

File: go.mod [hunk 3]
```diff
@@ -251,4 +251,6 @@ require (
 	lukechampine.com/blake3 v1.3.0 // indirect
 )
 
-go 1.22
+go 1.22.0
```
Comment (gammazero) [inline go.mod:254]: Seems too specific. I think we want `go 1.22`

**Pull request merged**